### ISSUE TITLE
Runtime Warning: "Publishing changes from within view updates is not allowed" when using .onReceive with ObservableObject in iOS 16

### ISSUE THREAD

**Issue by avasveliya** [Open-Bytes/SwiftUIFormValidator#22]
We are encountering the following issue in iOS 16 when using the .onReceive modifier with an ObservableObject:

```
.onReceive(form.manager.$allValid) { isValid in
      self.isSaveDisabled = !isValid
}
```

The error message displayed is:
**Publishing changes from within view updates is not allowed, this will cause undefined behavior.**

As per our understanding, in iOS 16, it is required for the ObservableObject class to be annotated with @MainActor in order to avoid such issues.

**Steps to Reproduce:**
- Use the code snippet above where form.manager.$allValid is being observed.
- Observe the behavior when this code executes.

**Expected Behavior:** The code should function without throwing the error, and changes should be made to isSaveDisabled without any warnings or undefined behavior.

**Actual Behavior:** We are seeing the error message: "Publishing changes from within view updates is not allowed, this will cause undefined behavior."

Please refer to the attached screenshot for the position where this error is being triggered.

![Image](https://github.com/user-attachments/assets/5d9e0dcd-0ead-4bfb-93db-169f41cd009e)

**Comment by MalekKamel** [comment]
Thanks for providing these detailed steps and the explanation.

I've gone ahead and applied the main actor logic everywhere to address the issue.
Could you please test this fix on version `1.2.2` and let me know if the error is resolved?

**Comment by avasveliya** [comment]
Thanks for the quick fix, it works as expected.
